frontend/src/components/AdminBookForm.js

```diff
@@ -0,0 +1,294 @@
+              <Label>Year</Label>
+              <Input
+                type="text"
+                name="year"
+                className={
+                  'form-control ' +
+                  (this.state.year !== '' &&
+                    (isNaN(this.state.year) ? 'is-invalid' : 'is-valid'))
+                }
+                onChange={this.handleChange}
+                maxLength="4"
+                pattern="[0-9]{4}"
+                required
+                placeholder="Ex: 2018"
+                value={this.state.year}
+              />
+              <FormFeedback invalid="true">
+                The year has to be a number.
+              </FormFeedback>
+            </Col>
+            <Col lg="6" className="year-grade">
+              <Label>Grade</Label>
+              <Input
+                type="text"
+                name="grade"
+                className={
+                  'form-control ' +
+                  (this.state.grade !== '' &&
+                    (isNaN(this.state.grade) ? 'is-invalid' : 'is-valid'))
+                }
+                maxLength="2"
+                pattern="[0-9]{1,2}"
+                onChange={this.handleChange}
+                placeholder="Ex: 8"
+                value={this.state.grade}
+              />
+              <FormFeedback invalid="true">
+                The grade has to be a number.
+              </FormFeedback>
+            </Col>
+          </Row>
+        </FormGroup>
+
+        <FormGroup>
+          <Label>Book Image URL</Label>
+          <Input
+            type="text"
+            name="cover_url"
+            className={
+              'form-control ' +
+              (this.state.coverURLValid !== null &&
+                this.state.cover_url !== '' &&
+                (this.state.coverURLValid ? 'is-valid' : 'is-invalid'))
+            }
+            onBlur={this.testImage}
+            onChange={this.handleChange}
+            value={this.state.cover_url}
+            placeholder="Ex: http://google.com/file.png"
+          />
+          <FormFeedback invalid="true">
+            We're having trouble loading that image.
+          </FormFeedback>
+          <FormFeedback valid>Image looks good!</FormFeedback>
+        </FormGroup>
+
+        <FormGroup>
+          <Label>Reader URL</Label>
+          <Input
+            type="text"
+            className={
+              'form-control ' +
+              (this.state.reader_url !== '' &&
+                (this.state.bookURLValid ? 'is-valid' : 'is-invalid'))
+            }
+            name="reader_url"
+            onChange={this.testBookURL}
+            value={this.state.reader_url}
+            placeholder="Ex: http://book.com/file.pdf"
+          />
+          <FormFeedback invalid="true">
+            That doesn't look like a valid link.
+          </FormFeedback>
+          <FormFeedback valid>That link looks good!</FormFeedback>
+        </FormGroup>
+
+        <FormGroup>
+          <Button
```

Comment:
Nice reuse :recycle:  for conditional rendering of button based on editing or creating book